bindings/rust/s2n-tls-tokio/tests/send_and_recv.rs

```diff
@@ -32,6 +32,28 @@ async fn send_and_recv_basic() -> Result<(), Box<dyn std::error::Error>> {
     Ok(())
 }
 
+#[tokio::test]
+async fn send_and_recv_into_vec() -> Result<(), Box<dyn std::error::Error>> {
+    let (server_stream, client_stream) = common::get_streams().await?;
+
+    let connector = TlsConnector::new(common::client_config()?.build()?);
+    let acceptor = TlsAcceptor::new(common::server_config()?.build()?);
+
+    let (mut client, mut server) =
+        common::run_negotiate(&connector, client_stream, &acceptor, server_stream).await?;
+
+    client.write_all(TEST_DATA).await?;
+
+    let mut received = vec![];
```

Comment:
Wonder why this works since here we are creating a vec of len 0.

In `poll_recv_uninitialized` we call `let buf_len: isize = buf.len()...` to get the len of the buf and attempt to read that amount `s2n_recv(conn..., buf_ptr, buf_len, ...)`.

So I would expect it to only attempt to read 0 bytes and never make progress. What am i missing?

Reply:
the `read_buf` grows the vec as needed

https://docs.rs/tokio/latest/tokio/io/trait.AsyncReadExt.html#method.read_buf

Reply:
Spoke offline. The underlying [BufMut impl](https://docs.rs/bytes/latest/src/bytes/buf/buf_mut.rs.html#1444-1446) grows the vec if it doesnt have capacity.

```
fn chunk_mut(&mut self) -> &mut UninitSlice {
        if self.capacity() == self.len() {
            self.reserve(64); // Grow the vec
        }
        ...
```